### PR TITLE
Review diagnostic messages for missing single quotes around symbol names

### LINKED ISSUE #1714: Audit diagnostic messages for missing quotes around symbol names
I think quite a few of our diagnostic messages were created by an automated tool that ported messages from old FxCop implementation. For example, see https://github.com/dotnet/roslyn-analyzers/issues/1713#issuecomment-395051433.

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #1714 

I did a first review of all the diagnostic messages to find missing quotes. I have deliberately skipped all description messages (let me know if you want me to update them).

I have, for now, only changed the resource files and when we agree upon the texts I'll update the various translation files.

Doing this review of the messages, I have a couple of remarks to make:

- There is no consistency on whether title and messages shall end (or not) with a period.

- Some of the messages are using a pascal case rule for all the words of the message, this probably reads weird when we happen to have the issue raised.

- There is no consistency on how attributes are referred to. We can find `Use [XXX]`, `Use [XXXAttribute]`, `Add XXX attribute` or `Add XXXAttribute`. It might be good to always use the same style.

### CODE REVIEW

**Comment by mavasani** [comment]
> There is no consistency on whether title and messages shall end (or not) with a period.

@sharwell do we have any precedence on this?

**Comment by mavasani** [comment]
> There is no consistency on how attributes are referred to.

I'd say lets just choose one style and be consistent. I am fine with using `Add XXXAttribute`.

File: src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/MicrosoftCodeQualityAnalyzersResources.resx [hunk 1]
```diff
@@ -154,13 +154,13 @@
     <value>Don't Store Async Lambdas as Void Returning Delegate Types</value>
   </data>
   <data name="PropagateCancellationTokensWhenPossibleTitle" xml:space="preserve">
-    <value>Propagate CancellationTokens When Possible</value>
+    <value>Propagate 'CancellationTokens' When Possible</value>
```
Comment (mavasani) [inline src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/MicrosoftCodeQualityAnalyzersResources.resx]:
```suggestion
    <value>Propagate 'CancellationToken' when possible</value>
```

File: src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/MicrosoftCodeQualityAnalyzersResources.resx [hunk 2]
```diff
@@ -154,13 +154,13 @@
...
   </data>
   <data name="PropagateCancellationTokensWhenPossibleDescription" xml:space="preserve">
     <value>#N/A</value>
   </data>
   <data name="PropagateCancellationTokensWhenPossibleMessage" xml:space="preserve">
-    <value>Propagate CancellationTokens When Possible</value>
+    <value>Propagate 'CancellationTokens' When Possible</value>
```
Comment (mavasani) [inline src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/MicrosoftCodeQualityAnalyzersResources.resx]:
```suggestion
    <value>Propagate 'CancellationToken' when possible</value>
```

File: src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/MicrosoftCodeQualityAnalyzersResources.resx (hunk 1 above)
Comment (Evangelink) [reply src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/MicrosoftCodeQualityAnalyzersResources.resx]: That's one example of
> Some of the messages are using a pascal case rule for all the words of the message

File: src/Microsoft.CodeAnalysis.Analyzers/Core/CodeAnalysisDiagnosticsResources.resx [hunk 3]
```diff
@@ -121,34 +121,34 @@
...
   <data name="AddLanguageSupportToAnalyzerTitle" xml:space="preserve">
     <value>Recommend adding language support to diagnostic analyzer.</value>
   </data>
   <data name="AddLanguageSupportToAnalyzerDescription" xml:space="preserve">
-    <value>Diagnostic analyzer is marked as supporting only one language, but the analyzer assembly doesn't seem to refer to any language specific CodeAnalysis assemblies, and so is likely to work for more than one language. Consider adding an additional language argument to DiagnosticAnalyzerAttribute.</value>
+    <value>Diagnostic analyzer is marked as supporting only one language, but the analyzer assembly doesn't seem to refer to any language specific CodeAnalysis assemblies, and so is likely to work for more than one language. Consider adding an additional language argument to 'DiagnosticAnalyzer' attribute.</value>
   </data>
   <data name="ApplyDiagnosticAnalyzerAttribute_1" xml:space="preserve">
-    <value>Apply DiagnosticAnalyzer attribute for '{0}'.</value>
+    <value>Apply 'DiagnosticAnalyzer' attribute for '{0}'.</value>
```
Comment (sharwell) [inline src/Microsoft.CodeAnalysis.Analyzers/Core/CodeAnalysisDiagnosticsResources.resx]: 💭 Another option here is:

```suggestion
    <value>Apply [DiagnosticAnalyzer] for '{0}'.</value>
```

**Comment by sharwell** [comment]
> There is no consistency on whether title and messages shall end (or not) with a period.

My preference is:

1. No '.' for single-sentence titles and messages (and avoid multi-sentence unless it substantially detracts from clarity)
2. Use full sentences with '.' for descriptions

**Comment by sharwell** [comment]
> There is no consistency on how attributes are referred to.

My preference is `'SomeAttribute'` for diagnostics, and `[Some]` for code fixes.

File: src/Microsoft.CodeAnalysis.Analyzers/Core/CodeAnalysisDiagnosticsResources.resx (hunk 3 above)
Comment (Evangelink) [reply src/Microsoft.CodeAnalysis.Analyzers/Core/CodeAnalysisDiagnosticsResources.resx]: I was thinking about this syntax first but then I thought that was too C# oriented (thinking about vb.net devs). Let me know if you still prefer to go for the brackets.

**Comment by Evangelink** [comment]
I will create a ticket and list what was said in this PR, close this PR and open a new one with only the changes we agreed upon.

**Pull request closed without merging**